src/modules/10-common/pages/404/NotFoundPage.tsx

```diff
@@ -22,7 +26,9 @@ export default function NotFoundPage(): JSX.Element {
       <Layout.Vertical spacing="large" flex={{ align: 'center-center' }}>
         <Heading>404</Heading>
         <Text>Oops, we could not find this page.</Text>
-        <Link to={CDS_NAV_2_0 ? routes.toMode({ module }) : '/'}>Go to Home</Link>
+        <Link to={props.redirectTo ? (CDS_NAV_2_0 ? props.redirectTo : routes.toMode({ module })) : '/'}>
```

Comment:
Better to make this independent of our FF as this can be used for other general purpose also

```suggestion
        <Link to={props.redirectTo ? props.redirectTo :  CDS_NAV_2_0 ? routes.toMode({ module }) : '/'}>
```